packages/mui-material-next/src/InputBase/InputBase.tsx

```diff
@@ -0,0 +1,727 @@
+  const ownerState = {
+    ...props,
+    color: colorProp || 'primary',
+    disabled: disabledState,
+    endAdornment,
+    error: errorState,
+    focused: focusedState,
+    formControl: muiFormControl,
+    fullWidth,
+    hiddenLabel: false, // TODO: material-next/FormControl integration
+    multiline,
+    size: sizeProp,
+    startAdornment,
+    type,
+  };
+
+  const classes = useUtilityClasses(ownerState);
+
+  const propsToForwardToInputSlot = {
+    'aria-describedby': ariaDescribedby,
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledby,
+    autoComplete,
+    autoFocus,
+    id,
+    onKeyDown,
+    onKeyUp,
+    name,
+    placeholder,
+    readOnly,
+    type,
+  };
+
+  const Root = slots.root || InputBaseRoot;
+  const rootProps: WithOptionalOwnerState<InputBaseRootSlotProps> = useSlotProps({
+    elementType: Root,
+    getSlotProps: getRootProps,
+    externalSlotProps: slotProps.root,
+    externalForwardedProps: other,
+    additionalProps: {
+      ref: forwardedRef,
+    },
+    ownerState,
+    className: [classes.root, className],
+  });
+
+  const InputComponent = multiline
+    ? slots.textarea ?? TextareaAutosize
+    : slots.input ?? InputBaseInput;
+
+  const {
+    ownerState: inputOwnerState,
+    ...otherInputProps
+  }: WithOptionalOwnerState<InputBaseInputSlotProps> = useSlotProps({
+    elementType: InputComponent,
+    getSlotProps: (otherHandlers: EventHandlers) => {
+      return getInputProps({
+        ...propsToForwardToInputSlot,
+        ...otherHandlers,
+      });
+    },
+    externalSlotProps: slotProps.input,
+    additionalProps: {
+      rows: multiline ? rows : undefined,
+      ...(multiline &&
+        !isHostComponent(InputComponent) && {
+          minRows: rows || minRows,
+          maxRows: rows || maxRows,
+        }),
+    },
+    ownerState,
+    className: classes.input,
+  });
+
+  if (process.env.NODE_ENV !== 'production') {
+    if (multiline) {
+      if (rows) {
+        if (minRows || maxRows) {
+          console.warn(
+            'MUI: You can not use the `minRows` or `maxRows` props when the input `rows` prop is set.',
+          );
+        }
+      }
+    }
+  }
+
+  const handleAutoFill = (event: React.AnimationEvent) => {
+    // Provide a fake value as Chrome might not let you access it for security reasons.
+    checkDirty(event.animationName === 'mui-auto-fill-cancel' ? inputRef : { value: 'x' });
+  };
+
+  // Check the input state on mount, in case it was filled by the user
+  // or auto filled by the browser before the hydration (for SSR).
+  React.useEffect(() => {
+    checkDirty(inputRef);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  let inputProps: WithOptionalOwnerState<InputBaseInputSlotProps> = {
+    ...otherInputProps,
+  };
+
+  // manually stop passing ownerState to TextareaAutosize
```

Comment:
Can we use `useSlotProps` for this?

Reply:
@DiegoAndai Some trickery was needed because `TextareaAutosize` renders a `Fragment`, but the props are actually going to the textarea DOM element [here](https://github.com/mui/material-ui/blob/master/packages/mui-base/src/TextareaAutosize/TextareaAutosize.tsx#L256)

I just found `ClickawayListener` is using a different hack to do the same thing [here](https://github.com/mui/material-ui/blob/master/packages/mui-base/src/Snackbar/Snackbar.tsx#L95-L96) 🙃

Perhaps it's time to update [this condition](https://github.com/mui/material-ui/blob/master/packages/mui-base/src/utils/appendOwnerState.ts#L45) in `appendOwnerState` to account for Fragments as well ~ CC @michaldudak

Reply:
Could we manually provide the type?

```ts
useSlotProps({
    elementType: InputComponent === TextareaAutosize ? 'textarea' : InputComponent
    // ...
});
```

Reply:
Could we manually provide the type?

```ts
useSlotProps({
    elementType: InputComponent === TextareaAutosize ? 'textarea' : InputComponent
    // ...
});
```

Reply:
👌 it works (I added a comment)